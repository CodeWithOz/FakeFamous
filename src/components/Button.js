import React from 'react';

const Button = props => {
  return (
    <div
      className={`ui basic ${props.color} button`}
      onClick={props.onClick}
    >{props.text}</div>
  );
};

Button.defaultProps = {
  color: 'green'
};

export default Button;
