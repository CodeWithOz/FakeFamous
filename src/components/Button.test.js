import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

test('Button component exists', () => {
  const button = renderer.create(
    <Button
      onClick={() => console.log('Dummy handler')}
      color='green'
      text='Dummy button'
    />
  );

  // render the component
  const tree = button.toJSON();
  expect(tree).toMatchSnapshot();
});
