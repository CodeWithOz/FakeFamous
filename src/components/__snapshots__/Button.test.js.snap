// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component exists 1`] = `
<div
  className="ui basic green button"
  onClick={[Function]}
>
  Dummy button
</div>
`;
